ruby/bindings/InitRubyBindings.cpp

```diff
@@ -920,6 +945,37 @@ void locateEmbeddedGems(bool use_bundler) {
 
     $logger.info "Bundling with groups [#{keep_groups.join(',')}]"
 
+    # Filter out dependencies with native extensions that we already have embedded in the CLI and that satisfy the requirements
+    locked_specs = Bundler.definition.instance_variable_get(:@locked_specs)
+
+    embedded_gems_to_activate.each do |spec|
+      next if spec.extensions.empty?
+
+      next unless locked_specs.any? {|locked_spec| locked_spec.name == spec.name}
```

Comment:
Alright, main dish starts here. Apologies in advance for what you're going to witness...


Step 1, grab the locked_specs, which is a private instance variable, and is the key one that bundler will use to resolve.

We start by looping on the emedded_gems_to_activate, and we keep only:
* Gems with native extensions
* Gems actually used by bundler as well.